pkg/config/remote/uptane/client_state.go

```diff
@@ -0,0 +1,126 @@
+package uptane
+
+// State represents the state of an uptane client
+type State struct {
+	ConfigState     map[string]MetaState
+	ConfigUserState map[string]MetaState
+	DirectorState   map[string]MetaState
+	TargetFilenames map[string]string
+}
+
+// MetaState represents the state of a tuf file
+type MetaState struct {
+	Version uint64
+	Hash    string
+}
+
+// ConfigRootVersion returns the version of the config root.json file
+func (s *State) ConfigRootVersion() uint64 {
+	meta, found := s.ConfigState[metaRoot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// ConfigSnapshotVersion returns the version of the config snapshot.json file
+func (s *State) ConfigSnapshotVersion() uint64 {
+	meta, found := s.ConfigState[metaSnapshot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// ConfigUserRootVersion returns the version of the user config root.json file
+func (s *State) ConfigUserRootVersion() uint64 {
+	meta, found := s.ConfigUserState[metaRoot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// ConfigUserSnapshotVersion returns the version of the user config snapshot.json file
+func (s *State) ConfigUserSnapshotVersion() uint64 {
+	meta, found := s.ConfigUserState[metaSnapshot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// DirectorRootVersion returns the version of the director root.json file
+func (s *State) DirectorRootVersion() uint64 {
+	meta, found := s.DirectorState[metaRoot]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// DirectorTargetsVersion returns the version of the director targets.json file
+func (s *State) DirectorTargetsVersion() uint64 {
+	meta, found := s.DirectorState[metaTargets]
+	if !found {
+		return 0
+	}
+	return meta.Version
+}
+
+// State returns the state of the uptane client
+func (c *Client) State() (State, error) {
+	c.Lock()
+	defer c.Unlock()
+	s := State{
+		ConfigState:     map[string]MetaState{},
+		ConfigUserState: map[string]MetaState{},
+		DirectorState:   map[string]MetaState{},
+		TargetFilenames: map[string]string{},
+	}
+
+	configMetas, err := c.datadogConfig.localStore.GetMeta()
+	if err != nil {
+		return State{}, err
+	}
+	for metaName, content := range configMetas {
+		version, err := metaVersion(content)
+		if err == nil {
```

Comment:
errors are being ignored, they are not even logged or reported. Is that on purpose? (several other occurrences in this file)

Reply:
@coignetp do you remember why ?

Reply:
hmmm you're right, @coignetp do you remember why you did this ?

Reply:
The content comes from the local store and was already validated, but I guess we could still log errors here

Reply:
I think we should return error regardless if the local state is borked